app/utils/vrf_key.go

```diff
@@ -206,20 +206,15 @@ func LoadVRFKey(keyFilePath string) (*VRFKey, error) {
 // If loadPath is specified, it loads the VRF key file at the specified
 // path. Otherwise, it generates a new VRF key, whose entropy is randomly
 // generated or obtained from the mnemonic, if provided.
-func LoadOrGenVRFKey(config *cfg.Config, loadPath, mnemonic string) (vrfPubKey sdkcrypto.PubKey, err error) {
+func LoadOrGenVRFKey(config *cfg.Config, loadPath string) (vrfPubKey sdkcrypto.PubKey, err error) {
```

Comment:
I'm not 100% sure how the dependency between the PKR and other modules is going to look. I thought the pubkeys was going to generate all required keys and the modules that utilise them 'just' send data to the module to have it signed with the correct key.